File: qc-core/src/main/java/com/eager/questioncloud/core/domain/library/model/UserQuestion.java
package com.eager.questioncloud.core.domain.library.model;

import com.eager.questioncloud.core.domain.question.model.Question;
import java.time.LocalDateTime;
import java.util.List;
import java.util.stream.Collectors;
import lombok.Builder;
import lombok.Getter;

@Getter
public class UserQuestion {
    private Long id;
    private Long userId;
    private Question question;
    private Boolean isUsed;
    private LocalDateTime createdAt;

    @Builder
    public UserQuestion(Long id, Long userId, Question question, Boolean isUsed, LocalDateTime createdAt) {
        this.id = id;
        this.userId = userId;
        this.question = question;
        this.isUsed = isUsed;
        this.createdAt = createdAt;
    }

    public static List<UserQuestion> create(Long userId, List<Question> questions) {
        LocalDateTime now = LocalDateTime.now();
        return questions
            .stream()
            .map(question -> UserQuestion
                .builder()
                .userId(userId)
                .question(question)
                .isUsed(false)
                .createdAt(now)
                .build())
            .collect(Collectors.toList());
    }
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/userquestion/model/UserQuestion.kt
package com.eager.questioncloud.core.domain.userquestion.model

import java.time.LocalDateTime
import java.util.stream.Collectors

class UserQuestion(
    val id: Long? = null,
    val userId: Long,
    val questionId: Long,
    val isUsed: Boolean = false,
    val createdAt: LocalDateTime = LocalDateTime.now(),
) {
    companion object {
        fun create(userId: Long, questionIds: List<Long>): List<UserQuestion> {
            return questionIds
                .stream()
                .map { questionId: Long ->
                    UserQuestion(userId = userId, questionId = questionId)
                }
                .collect(Collectors.toList())
        }
    }
}
